qualtran/_infra/adjoint.py

```diff
@@ -147,7 +147,7 @@ def decompose_from_registers(
         self, *, context: cirq.DecompositionContext, **quregs: NDArray[cirq.Qid]
     ) -> cirq.OP_TREE:
         if isinstance(self.subbloq, GateWithRegisters):
-            return cirq.inverse(self.subbloq.decompose_from_registers(context=context, **quregs))
+            return cirq.inverse(*self.subbloq.decompose_from_registers(context=context, **quregs))
```

Comment:
I don't think you should do this.
```suggestion
            return cirq.inverse(self.subbloq.decompose_from_registers(context=context, **quregs))
```